Content.Server/Weapon/Melee/Esword/EswordSystem.cs

```diff
@@ -0,0 +1,123 @@
+using System.Collections.Generic;
+using Content.Shared.Item;
+using Content.Server.Tools.Components;
+using Content.Shared.Interaction;
+using Content.Shared.ActionBlocker;
+using Content.Shared.FixedPoint;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Player;
+
+namespace Content.Server.Weapon.Melee.Esword
+{
+    internal class EswordSystem : EntitySystem
+    {
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<EswordComponent, MeleeHitEvent>(OnMeleeHit);
+            SubscribeLocalEvent<EswordComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<EswordComponent, InteractUsingEvent>(OnInteractUsing);
+
+        }
+
+        private void OnMeleeHit(EntityUid uid, EswordComponent comp, MeleeHitEvent args)
+        {
+            
+            if (comp.Activated == true)
+            {
+                args.BonusDamage.DamageDict = new Dictionary<string, FixedPoint2>()
+                {
+                    {"Slash", FixedPoint2.New(25)},
+                };
```

Comment:
The damage itself should be specified as a yaml data-field, not hardcoded.

Reply:
Also, surely this should (also) be dealing heat damage?

Reply:
Oh yeah, fuck, you're right. I'm gonna split the total damage into Slash and Burn, that makes more sense than straight Slash.
On another note, what should I look for as an example of how to implement damage changes based on if an item is activated or not?
On the .yml I did Blunt 5 (if you just slap someone with handle it should do some damage), but if you activate it, it does the actual damage it's supposed to.
I had a hard time finding another example on how to do this without manually forcing the DamageDict like this.

Reply:
> On another note, what should I look for as an example of how to implement damage changes based on if an item is activated or not?

AFAIK there aren't really any weapons that have that functionality atm. The `BonusDamage` stuff was just added for the sake of welders welders, where they deal burn damage if ignited (see `OnMeleeHit()` in `ToolSystem.Welder.cs`.

But there's some jank in how that works. Because the event basically returns a bonus & modifier, there is no easy way to just overwrite the damage values. So for example, the welders `litDamageBonus` in yaml files is +10 burn and -10 blunt, because the welder does 10 blunt damage by default, but it shouldn't when being used as a burning tool.

In hindsight, instead of having the `MeleeHitEvent` return a bonus & modifier, maybe it should just directly pass along the current damage value, so that each systems can directly modify it, including completely overwriting it.

Reply:
Also on the damage values, remember that 5-blunt is the same as unarmed punching, so even 6-7 is probably fine. Most tools like wrenches & crowbars do 10 blunt damage. The welder should probably have been a tad less than 10, but i guess its still pretty bulky.